app/src/main/java/com/github/epfl/meili/forum/MockPostService.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.epfl.meili.forum
+
+import android.util.Log
+import com.github.epfl.meili.forum.Post.Companion.toPost
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.tasks.await
+import java.util.*
+import kotlin.collections.ArrayList
+
+class MockPostService : PostService(){
+    private var idCounter: Int = 0
+    public var posts: ArrayList<Post> = ArrayList()
+    private val TAG = "MockPostService"
+    private val FAKE_ID = "FAKE_ID"
+    private val FAKE_AUTHOR = "FAKE_AUTHOR"
+    private val FAKE_TITLE = "FAKE_TITLE"
+    private val FAKE_TEXT = "FAKE_TEXT"
+    private var observers: ArrayList<Observer> = ArrayList()
+
+
+    init {
+            posts.add(Post(FAKE_ID, FAKE_AUTHOR, FAKE_TITLE, FAKE_TEXT))
+            setChanged() // Tell Observable that its state has changed
+            notifyObservers()
+        }
+
+    override suspend fun getPostFromId(id: String?): Post? {
+        for(post:Post  in posts){
+            if(post.id == id){
+                return post;
+            }
+        }
+        return null
+    }
+
+    override suspend fun getPosts(): List<Post> {
+        return posts
+    }
+
+    override fun addPost(author: String, title: String, text: String) {
+        posts.add(Post((idCounter++).toString(),author, title, text))
+        notifyObservers()
+    }
+
+    override fun addObserver(o: Observer?) {
+        super.addObserver(o)
+        if(o != null && !observers.contains(o)){
+            observers.add(o)
+        }
+        notifyObservers()
+    }
+
+    override fun notifyObservers() {
+        super.notifyObservers()
+
+        for(observer in observers){
+            observer.update(this, posts)
+        }
+    }
+}
```

Comment:
Conversation from previous PR :  

@MarcelTorne :
Instead of implementing the observer methods here, don't you think it would be better to have it in PostService? So that the children can inherit it?

@Bwananah :
This is an Observable method. The changes to the Observable methods only apply to the Mock Service, so I think it is better there. PostService already inherits from Observable(), so FirebasePostService just uses the Observable() default methods.

@MarcelTorne :
Sorry, maybe I am not understanding correctly, but I don't really understand why you would use default methods from Observable in FirebasePostService but not in the mock service. Could you clarify a little bit more, please? Thanks!

@Bwananah :
I'm not quite sure, @ahmedc3ca did this part. But I tried what you said and the mock service did not end up working.

Reply:
For context: when I made the MockPostService, I created the PostService interface along with it. The FirebasePostService was already there. I didn't want PostService to implement the observer methods because then I would have to add the observables list to it. This means that FirebasePostService will also use all of that, but it didn't seem to me that Ewan was using them. So from my point of view, the interface PostService didn't necessarily use the observer's methods. If you think it is cleaner to have the interface implement the observer methods tho, it is easily arrangeable (just copy/paste from MockPostService).

Reply:
Then maybe the best would be to remove the Observable interface from PostService? Because if you say that FirebasePostService is not using it maybe we don't need it. Otherwise, if you see that you do need it then I think the best would be to implement the methods in PostService

Reply:
FirebasePostService is using the Observable interface. It uses notifyObservers() and setChanged()

Reply:
I have moved the Observable changes into PostService